### PR TITLE
KDESKTOP-1330-Remove-hasRight-method-from-Executor

### PR DESCRIPTION
**Pull request by ClementKunz**
Following https://github.com/Infomaniak/desktop-kDrive/pull/359, the method `hasRight` in Executor is not needed anymore. Remove it but keep the logic on special folders ("Common documents" and "Shared")

### CODE REVIEW

File: src/libsyncengine/propagation/executor/executorworker.cpp [hunk 1]
```diff
@@ -1420,90 +1412,25 @@ ExitInfo ExecutorWorker::generateDeleteJob(SyncOpPtr syncOp, bool &ignored, bool
...
-                if (!exists) {
-                    LOGW_SYNCPAL_WARN(_logger,
-                                      L"Item: " << Utility::formatSyncPath(absoluteLocalFilePath) << L" doesn't exist anymore!");
-                    return false;
-                }
+        if (!newCorrespondingParentNode || !newCorrespondingParentNode->id().has_value()) {
+            return false;
+        }
 
-                if (!correspondingNode->id().has_value()) {
```
[the remark this answers is not in the document]
Comment (ClementKunz) [reply src/libsyncengine/propagation/executor/executorworker.cpp:1493]: Good question actually.
On ExecutorWorker::generateEditJob, line 969 : `if (!syncOp->correspondingNode()->id()) {`
And it looks like it is not needed anywhere else.

File: src/libsyncengine/propagation/executor/executorworker.cpp [hunk 2]
```diff
@@ -1788,9 +1715,12 @@ ExitInfo ExecutorWorker::handleForbiddenAction(SyncOpPtr syncOp, const SyncPath
         case OperationType::Create: {
             cancelType = CancelType::Create;
             ignored = true;
-            // If the rename failed (the file doesn't exist), it is a forbiden download, we should keep it in DB.
-            removeFromDb = PlatformInconsistencyCheckerUtility::renameLocalFile(
-                    absoluteLocalFilePath, PlatformInconsistencyCheckerUtility::SuffixType::Blacklisted);
+            if (!PlatformInconsistencyCheckerUtility::renameLocalFile(
+                        absoluteLocalFilePath, PlatformInconsistencyCheckerUtility::SuffixType::Blacklisted)) {
+                LOGW_SYNCPAL_WARN(_logger, L"PlatformInconsistencyCheckerUtility::renameLocalFile: "
```
[the remark this answers is not in the document]
Comment (ClementKunz) [reply src/libsyncengine/propagation/executor/executorworker.cpp]: Done in 19d59cf9e26c91beb8178b16146e8926c34746e2

File: src/libsyncengine/propagation/executor/executorworker.cpp [hunk 3]
```diff
@@ -411,7 +416,12 @@ ExitInfo ExecutorWorker::handleCreateOp(SyncOpPtr syncOp, std::shared_ptr<Abstra
...
                     }
                 }
-                return exitInfo;
+                if (const ExitInfo exitInfo2 = handleForbiddenAction(syncOp, relativeLocalFilePath, ignored); !exitInfo2) {
+                    LOGW_SYNCPAL_WARN(_logger, L"Error in handleForbiddenAction for item: "
+                                                       << Utility::formatSyncPath(relativeLocalFilePath));
+                    return exitInfo2;
+                }
+                return {ExitCode::BackError, ExitCause::FileAccessError};
             }
```
[the remark this answers is not in the document]
Comment (ClementKunz) [reply src/libsyncengine/propagation/executor/executorworker.cpp:426]: You arre 100% right, thx for your vigilance: b3e24ee4222ff6f0910b862d5d21280714795e45

File: src/libsyncengine/propagation/executor/executorworker.cpp [hunk 4]
```diff
@@ -400,21 +405,28 @@ ExitInfo ExecutorWorker::handleCreateOp(SyncOpPtr syncOp, std::shared_ptr<Abstra
 
         if (job && syncOp->affectedNode()->type() == NodeType::Directory) {
             // Propagate the directory creation immediately in order to avoid blocking other dependant job creation
-            if (ExitInfo exitInfo = runCreateDirJob(syncOp, job); !exitInfo) {
```
Comment (herve-er) [inline src/libsyncengine/propagation/executor/executorworker.cpp:403]: Maybe the logic of `runCreateDirJob` could include this back error check. 
This is just a tough, here is a dedicated Jira card for a dedicated PR->
https://infomaniak.atlassian.net/jira/software/c/projects/KDESKTOP/boards/341

**Review by herve-er — APPROVED** [review]

**Review by ChristopheLarchier — APPROVED** [review]

**Pull request merged**